### PR TITLE
fix(breadcrumbs): Ensure maximum lengths on several attributes

### PR DESCRIPTION
**Pull request by dcramer**
- Trim message, category, and level attributes
- Update pytest support to work in other environments

Fixes GH-604

### CODE REVIEW

**Comment by dcramer** [comment]
This also fixes py.test support which had been broken for me locally.

flake8 is still broken (i had to commit with -n) but I dont have time to deal with that. We should be careful changing development/test environment, as it seems we might have done it without considering consequences (e.g. I assume we bumped the version of flake8 and it has never worked w/ our hooks since)

**Comment by dcramer** [comment]
@jan-auer @HazAT throwing you two on since we dont have an sdks team (afaik) on github

File: raven/breadcrumbs.py [hunk 1]
```diff
@@ -28,40 +32,59 @@ def event_payload_considered_equal(a, b):
 
 class BreadcrumbBuffer(object):
 
-    def __init__(self, limit=100):
+    def __init__(self, limit=100, message_max_length=1024):
```
Comment (ashwoods) [inline raven/breadcrumbs.py]: I'll set `limit` and `message_max_length` as settings in the new-settings-branch

**Comment by ashwoods** [comment]
@dcramer Thx!

**Pull request merged**